SCloudRandomizer/MainViewController.m

```diff
@@ -233,66 +268,44 @@ - (MBProgressHUD*) getProgressBar:(MBProgressHUDMode)progressHudMode
     return progressHud;
 }
 
+- (void)downloadTrack:(Track *)track
+          progressHud: (MBProgressHUD *) progressHud
+    completionHandler:(singleTrackDownloaded)completionHandler {
+    NSLog(@"The streamURL is: %@", track.streamUrl);
+    dispatch_async([Utility getMainQueue], ^{
+        self.scAudioStream = [track getStream];
+        if (completionHandler) completionHandler();
```

Comment:
I think we should just call completionHandler directly. We should fail hard if we call downloadTrack with a nil completion handler